### PR TITLE
Fix/bbb recordings server failure

### LINKED ISSUE #3889: Differentiate BBB recording-fetch failures from rooms with no recordings
## Context

  Follow-up to #1661, section **D) Recordings**.

  The BBB recordings popup does not reliably distinguish between:

  - A successful BBB response containing zero recordings.
  - A failure to fetch recordings because all eligible BBB servers are unavailable or return errors.

  This can cause the popup to keep loading or render an empty result without useful feedback.

  ## Expected behavior

  - Show **“No recordings are available.”** when BBB successfully returns zero recordings.
  - Show the existing recording-fetch failure message when every eligible BBB server fails.
  - Reuse the existing `bbb.failed` error code.
  - Do not introduce a new error code before centralized error handling is implemented.
  - Accept a successful response when only some eligible BBB servers fail.
  - Ensure the popup always finishes with recordings, an empty state, or an error state.

  ## Reproduction steps

  1. Open a BBB room with recordings permission.
  2. Open the recordings popup.
  3. Test with a healthy BBB server that returns no recordings.
  4. Test with all eligible BBB servers unavailable.
  5. Test with one unavailable server and one healthy server.

  ## Acceptance criteria

  - [ ] Successful empty BBB response displays “No recordings are available.”
  - [ ] All-server recording-fetch failure displays the existing BBB failure message.
  - [ ] Partial server failure still returns results from a healthy server.
  - [ ] Recording requests to eligible servers run concurrently.
  - [ ] The recordings popup does not spin indefinitely or render blank.
  - [ ] Backend behavior is covered by focused tests.

### PR DESCRIPTION
**Pull request by sarafarajnasardi**
## Description

Fixes the BBB recordings flow so the UI distinguishes between rooms with no recordings and failures while fetching recordings.

  Closes #3889 

  ## Changes

  - Fetch recordings from eligible BBB servers concurrently.
  - Return results when at least one BBB server responds successfully.
  - Reuse the existing bbb.failed error when every server fails or returns malformed data.
  - Display “No recordings are available.” for successful empty responses.
  - Prevent the recordings popup from spinning indefinitely or rendering blank.
  - Add focused tests for empty results, server failures, malformed responses, and partial server failures.

https://github.com/user-attachments/assets/74a4abff-b486-421f-9107-871d5b808b96

### CODE REVIEW

File: app/eventyay/webapp/video/src/components/RecordingsPrompt.vue [hunk 1]
```diff
@@ -28,19 +29,23 @@ export default {
...
+			recordings: [],
 			error: null,
+			loading: true,
 		}
 	},
 	computed: {},
 	async created() {
 		try {
-			this.recordings = (await api.call('bbb.recordings', {room: this.room.id}, {timeout: 150000})).results
+			const response = await api.call('bbb.recordings', {room: this.room.id})
```
Comment (Rachit7168) [inline app/eventyay/webapp/video/src/components/RecordingsPrompt.vue:40]: Just to confirm, was the removal of the {timeout: 150000} parameter intentional? Since the backend now enforces a 10-second timeout concurrently, dropping a 150-second frontend timeout makes sense, but I wanted to double-check that it wasn't accidentally omitted.

File: app/eventyay/base/services/bbb.py [hunk 2]
```diff
@@ -394,21 +395,39 @@ def _get_possible_servers(self):
 
     async def get_recordings_for_room(self, room):
         recordings = []
-        for server in await self._get_possible_servers():
+        call = await get_call_for_room(room)
```
Comment (Rachit7168) [inline app/eventyay/base/services/bbb.py:398]: Moving get_call_for_room(room) outside of the server loop is a great optimization! It saves unnecessary duplicate database queries that were previously happening in the sequential loop.

File: app/eventyay/base/services/bbb.py [hunk 3]
```diff
@@ -394,21 +395,39 @@ def _get_possible_servers(self):
...
+        recording_urls = [
+            get_url(
+                "getRecordings",
+                {"meetingID": call.meeting_id, "state": "any"},
+                server.url,
+                server.secret,
+            )
+            for server in servers
+        ]
+        responses = await asyncio.gather(
```
Comment (Rachit7168) [inline app/eventyay/base/services/bbb.py:413]: Excellent change to use asyncio.gather here! Fetching from all configured BBB servers concurrently (rather than sequentially) will significantly reduce the response time. Additionally, since self._get internally catches its exceptions and gracefully returns False, we avoid the pitfalls of unhandled exceptions crashing gather. This is very robust.

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
Please address

File: app/eventyay/webapp/video/src/components/RecordingsPrompt.vue (hunk 1 above)
Comment (sarafarajnasardi) [reply app/eventyay/webapp/video/src/components/RecordingsPrompt.vue:40]: yes it was intentional

**Review by Rachit7168 — APPROVED** [review]
LGTM , tested : 

Image

https://github.com/user-attachments/assets/507e286f-3180-40b9-a037-ab7f728504cf

https://github.com/user-attachments/assets/6f8aa38a-aab2-4072-9521-afb253c14c32

**Review by Saksham-Sirohi — APPROVED** [review]
